x/quarantine/client/cli/util.go

```diff
@@ -0,0 +1,103 @@
+//
+// This validation is (hopefully) already done by the node, but it's more
+// user-friendly to also do it here, before a request is actually sent.
+func validateAddress(addr string, argName string) (string, error) {
+	if _, err := sdk.AccAddressFromBech32(addr); err != nil {
+		return "", sdkerrors.ErrInvalidAddress.Wrapf("invalid %s: %v", argName, err)
+	}
+	return addr, nil
+}
+
+// ParseAutoResponseUpdatesFromArgs parses the args to extract the desired AutoResponseUpdate entries.
+// The args should be the entire args list. Parsing of the auto-response updates args will start at startIndex.
+func ParseAutoResponseUpdatesFromArgs(args []string, startIndex int) ([]*quarantine.AutoResponseUpdate, error) {
+	iLastArg := len(args) - 1 - startIndex // index of the last arg.
+	arArgCount := 0                        // a count of arguments that have been auto-responses.
+	arAddrCount := 0                       // a count of from_addresses provided for the most recent auto-response.
+	var lastArArg string                   // the actual arg string of the last auto-response arg.
+	var ar quarantine.AutoResponse         // the current auto-response.
+
+	var rv []*quarantine.AutoResponseUpdate
+
+	for i, arg := range args[startIndex:] {
+		newAr, isAr := ParseAutoResponseArg(arg)
+		// first arg must be an auto-response.
+		if i == 0 && !isAr {
+			return nil, fmt.Errorf("invalid arg %d: invalid auto-response: %q", i+startIndex+1, arg)
+		}
+		if isAr {
+			// If not the first arg, there must be at least one address for the previous auto-response.
+			if i > 0 && arAddrCount == 0 {
+				return nil, fmt.Errorf("invalid arg %d: no from_address args provided after auto-response %d: %q", i+startIndex+1, arArgCount, lastArArg)
+			}
+			// The last argument cannot be an auto-response either.
+			if i == iLastArg {
+				// Slightly different message on purpose. Makes it easier to track down the source of an error.
+				return nil, fmt.Errorf("invalid arg %d: last arg cannot be an auto-response, got: %q", i+startIndex+1, arg)
+			}
+			arArgCount += 1
+			ar = newAr
+			lastArArg = arg
+			arAddrCount = 0
+		} else {
+			arAddrCount += 1
+			fromAddr, err := validateAddress(arg, "from_address")
+			if err != nil {
+				return nil, fmt.Errorf("unknown arg %d %q: auto-response %d %q: from_address %d: %w", i+startIndex+1, arg, arArgCount, lastArArg, arAddrCount, err)
+			}
+			rv = append(rv, &quarantine.AutoResponseUpdate{
+				FromAddress: fromAddr,
+				Response:    ar,
+			})
+		}
+	}
+
+	return rv, nil
+}
+
+// ParseAutoResponseArg converts the provided arg to an AutoResponse enum entry.
+// The bool return value is true if parsing was successful.
+func ParseAutoResponseArg(arg string) (quarantine.AutoResponse, bool) {
+	switch strings.ToLower(arg) {
+	case "accept", "a", "1":
+		return quarantine.AUTO_RESPONSE_ACCEPT, true
+	case "decline", "d", "2":
+		return quarantine.AUTO_RESPONSE_DECLINE, true
+	case "unspecified", "u", "off", "o", "0":
```

Comment:
This goes with my previous comment I made on the specs. Remove `unspecified` and check for `off`. Also, change the return type to `AUTO_RESPONSE_OFF`.

```suggestion
	case "off", "o", "0":
```

Reply:
I'm going to mark this as "resolved" since the conversation about it is happening here: https://github.com/provenance-io/cosmos-sdk/pull/335#discussion_r1081843133

Once we have a conclusion there, I will be sure to also update this piece too.